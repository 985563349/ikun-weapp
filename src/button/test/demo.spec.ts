import path from 'node:path';
import simulate from 'miniprogram-simulate';

test('should render demo and match snapshot', async () => {
  const id = simulate.load(path.resolve(__dirname, '../demo/button'), {
    rootPath: path.resolve(__dirname, '../../'),
  });
  const container = simulate.render(id);

  container.attach(document.createElement('parent-wrapper'));
  expect(container.toJSON()).toMatchSnapshot();
});
