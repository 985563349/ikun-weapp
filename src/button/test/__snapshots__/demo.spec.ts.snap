// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render demo and match snapshot 1`] = `
<main>
  <wx-view>
    <wx-view>
      Button 按钮
    </wx-view>
    <wx-view>
      <wx-view>
        类型
      </wx-view>
      <wx-view>
        <k-button>
          Default
        </k-button>
        <k-button
          color="primary"
          variant="clear"
        >
          Clear
        </k-button>
        <k-button
          color="success"
          variant="outline"
        >
          Outline
        </k-button>
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        图标
      </wx-view>
      <wx-view>
        <k-button
          color="primary"
          loading="{{true}}"
        >
          Loading
        </k-button>
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        状态
      </wx-view>
      <wx-view>
        <k-button
          disabled="{{true}}"
        >
          Default
        </k-button>
        <k-button
          color="primary"
          disabled="{{true}}"
          variant="clear"
        >
          Clear
        </k-button>
        <k-button
          color="success"
          disabled="{{true}}"
          variant="outline"
        >
          Outline
        </k-button>
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        尺寸
      </wx-view>
      <wx-view>
        <k-button
          size="sm"
        >
          Small
        </k-button>
        <k-button>
          Default
        </k-button>
        <k-button
          size="lg"
        >
          large
        </k-button>
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        主题
      </wx-view>
      <wx-view>
        <k-button>
          Default
        </k-button>
        <k-button
          color="primary"
        >
          Primary
        </k-button>
        <k-button
          color="success"
        >
          Success
        </k-button>
        <k-button
          color="warning"
        >
          Warning
        </k-button>
        <k-button
          color="danger"
        >
          Dander
        </k-button>
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        形状
      </wx-view>
      <wx-view>
        <k-button
          color="danger"
          shape="round"
        >
          Success
        </k-button>
        <k-button
          color="warning"
          shape="round"
          variant="outline"
        >
          Success
        </k-button>
        <k-button
          color="success"
          expand="block"
          size="lg"
        >
          Primary
        </k-button>
        <k-button
          color="primary"
          expand="full"
          size="lg"
        >
          Success
        </k-button>
      </wx-view>
    </wx-view>
  </wx-view>
</main>
`;
